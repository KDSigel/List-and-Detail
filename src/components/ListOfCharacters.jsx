import { Link } from 'react-router-dom';

export default function ListOfCharacters({characters}) {
    return (
        <ul>
            {characters.map((character) => (
                <li>
                    <Link to={`/characters/${character.id}`}>
                    <img src={character.photoUrl} alt={character.name} width="200" />
                    <div>name: {character.name}</div>
                    </Link>
                </li>
            ))}
        </ul>
    )
}
