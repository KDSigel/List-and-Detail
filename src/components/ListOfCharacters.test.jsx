// add a fake list of characters

import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import ListOfCharacters from './ListOfCharacters'

it('tests a snapshot of Characters component', () => {

const characters = [{
    _id: 123,
    name: 'karl',
    position: 'driver',
    allies: 'cohort',
    photoUrl: 'https://en.wikipedia.org/wiki/Vasily_Arkhipov#/media/File:Vasili_Arkhipov_young.jpg'
}]

const {container} = render(
    <MemoryRouter>
        <ListOfCharacters characters={characters}/>
    </MemoryRouter>

)

expect(container).toMatchSnapshot()
}
)