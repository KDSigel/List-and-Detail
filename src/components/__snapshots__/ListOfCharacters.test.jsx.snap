// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests a snapshot of Characters component 1`] = `
<div>
  <ul>
    <li>
      <a
        href="/character/123"
      >
        <img
          alt="karl"
          src="https://en.wikipedia.org/wiki/Vasily_Arkhipov#/media/File:Vasili_Arkhipov_young.jpg"
          width="200"
        />
        <div>
          karl
        </div>
      </a>
    </li>
  </ul>
</div>
`;
